src/classlibnative/bcltype/system.cpp

```diff
@@ -225,17 +225,18 @@ FCIMPL1(Object*, SystemNative::_GetEnvironmentVariable, StringObject* strVarUNSA
     int len, newLen;
 
     // Get the length of the environment variable.
-    WCHAR dummy;    // prefix complains if pass a null ptr in, so rely on the final length parm instead
-    len = WszGetEnvironmentVariable(strVar->GetBuffer(), &dummy, 0);
+    PathString dummy;    // prefix complains if pass a null ptr in, so rely on the final length parm instead
+    len = WszGetEnvironmentVariable(strVar->GetBuffer(), dummy);
 
     while (len != 0)
     {
         // Allocate the string.
-        refRetVal = StringObject::NewString(len);
-
-        // Get the value.
-        newLen = WszGetEnvironmentVariable(strVar->GetBuffer(), refRetVal->GetBuffer(), len);
-        if (newLen != (len - 1))
+        refRetVal = StringObject::NewString(len + 1);
+ 
+        wcscpy_s(refRetVal->GetBuffer(), len + 1, dummy);
+        // Get the value Again.
+        newLen = WszGetEnvironmentVariable(strVar->GetBuffer(), dummy);
```

Comment:
No need for this now. In the original code we had to allocate the buffer first, and then get the environment variable. Now this is all done by WszGetEnvironmentVariable wrapper. Also don't need the surrounding while loop any more.

On the other hand, the if statement in GetEnvironmentVariableWrapper should really be a while loop, since the environment variable can be modified by another thread between two calls to GetEnvironmentVariableW.